include/circt/Dialect/FIRRTL/FIRRTLUtils.h

```diff
@@ -232,8 +232,12 @@ mapBaseTypeNullable(FIRRTLType type,
 
 /// Given a type, return the corresponding lowered type for the HW dialect.
 /// Non-FIRRTL types are simply passed through. This returns a null type if it
-/// cannot be lowered.
-Type lowerType(Type type);
+/// cannot be lowered. The optional function is required to specify how to lower
+/// AliasTypes.
+Type lowerType(
+    Type type, std::optional<Location> loc = {},
```

Comment:
It is necessary to use `std::optional`? Can we just use nullptr?

Reply:
I think Location can't be null, we need to use `optional`